unchanged:
--- ./src/tools/io/vpParallelPort.cpp	2012-06-15 14:12:11.489662035 -0700
+++ ./src/tools/io/vpParallelPort.cpp	2012-06-15 14:15:37.697535714 -0700
@@ -47,6 +47,7 @@
 #  include <sys/types.h>
 #  include <sys/stat.h>
 #  include <fcntl.h>
+#  include <unistd.h>
 #  include <sys/ioctl.h>
 
 #  include <visp/vpParallelPort.h>
unchanged:
--- ./src/tracking/klt/vpKltOpencv.h	2012-06-15 14:12:11.489662035 -0700
+++ ./src/tracking/klt/vpKltOpencv.h	2012-06-15 14:15:37.697535714 -0700
@@ -62,6 +62,7 @@
 #if (VISP_HAVE_OPENCV_VERSION >= 0x020101) // Require opencv >= 2.1.1
 #  include <opencv2/imgproc/imgproc.hpp>
 #  include <opencv2/video/tracking.hpp>
+#  include <opencv2/legacy/legacy.hpp>
 #else
 #  ifndef _EiC
 #    include "cv.h"
unchanged:
--- ./src/robot/real-robot/afma4/vpServolens.cpp	2012-06-15 14:12:07.170125512 -0700
+++ ./src/robot/real-robot/afma4/vpServolens.cpp	2012-06-15 14:13:00.284426422 -0700
@@ -58,6 +58,7 @@
 #include <fcntl.h>
 #include <stdlib.h>
 #include <string.h>
+#include <unistd.h>
 
 #include <visp/vpServolens.h>
 #include <visp/vpRobotException.h>
unchanged:
--- ./src/tracking/klt/vpKltOpencv.h	2012-06-15 14:12:10.769739280 -0700
+++ ./src/tracking/klt/vpKltOpencv.h	2012-06-15 14:16:01.065028275 -0700
@@ -61,6 +61,7 @@
 
 #if (VISP_HAVE_OPENCV_VERSION >= 0x020101) // Require opencv >= 2.1.1
 #  include <opencv2/imgproc/imgproc.hpp>
+#  include <opencv2/legacy/legacy.hpp>
 #  include <opencv2/video/tracking.hpp>
 #else
 #  ifndef _EiC
diff -r -u ./src/key-point/vpFernClassifier.h ./src/key-point/vpFernClassifier.h
--- ./src/key-point/vpFernClassifier.h	2011-09-20 16:15:14.000000000 +0200
+++ ./src/key-point/vpFernClassifier.h	2012-06-01 11:30:38.597237236 +0200
@@ -48,7 +48,11 @@
 #include <visp/vpConfig.h>
 
 #if (VISP_HAVE_OPENCV_VERSION >= 0x020000) // Require opencv >= 2.0.0
-#if (VISP_HAVE_OPENCV_VERSION >= 0x020101) // Require opencv >= 2.1.1
+#if (VISP_HAVE_OPENCV_VERSION >= 0x020400) // Require opencv >= 2.4.0
+#  include <opencv2/imgproc/imgproc.hpp>
+#  include <opencv2/features2d/features2d.hpp>
+#  include <opencv2/legacy/legacy.hpp>
+#elif (VISP_HAVE_OPENCV_VERSION >= 0x020101) // Require opencv >= 2.1.1
 #  include <opencv2/imgproc/imgproc.hpp>
 #  include <opencv2/features2d/features2d.hpp>
 #elif (VISP_HAVE_OPENCV_VERSION >= 0x020000) // Require opencv >= 2.0.0
diff -r -u ./src/key-point/vpKeyPointSurf.h ./src/key-point/vpKeyPointSurf.h
--- ./src/key-point/vpKeyPointSurf.h	2011-07-19 19:09:36.000000000 +0200
+++ ./src/key-point/vpKeyPointSurf.h	2012-06-01 11:31:05.168237904 +0200
@@ -61,7 +61,10 @@
 
 #if defined (VISP_HAVE_OPENCV) 
 
-#if (VISP_HAVE_OPENCV_VERSION >= 0x020101) // Require opencv >= 2.1.1
+#if (VISP_HAVE_OPENCV_VERSION >= 0x020400) // Require opencv >= 2.4.0
+#  include <opencv2/features2d/features2d.hpp>
+#  include <opencv2/legacy/compat.hpp>
+#elif (VISP_HAVE_OPENCV_VERSION >= 0x020101) // Require opencv >= 2.1.1
 #  include <opencv2/features2d/features2d.hpp>
 #elif (VISP_HAVE_OPENCV_VERSION >= 0x010100) // Require opencv >= 1.1.0
 #  include <cxcore.h>
diff -r -u ./src/key-point/vpPlanarObjectDetector.h ./src/key-point/vpPlanarObjectDetector.h
--- ./src/key-point/vpPlanarObjectDetector.h	2011-09-20 16:15:14.000000000 +0200
+++ ./src/key-point/vpPlanarObjectDetector.h	2012-06-01 11:31:32.365240341 +0200
@@ -47,7 +47,12 @@
 
 #if (VISP_HAVE_OPENCV_VERSION >= 0x020000) // Require opencv >= 2.0.0
 
-#if (VISP_HAVE_OPENCV_VERSION >= 0x020101) // Require opencv >= 2.1.1
+#if (VISP_HAVE_OPENCV_VERSION >= 0x020400) // Require opencv >= 2.4.0
+#  include <opencv2/imgproc/imgproc.hpp>
+#  include <opencv2/features2d/features2d.hpp>
+#  include <opencv2/calib3d/calib3d.hpp>
+#  include <opencv2/legacy/legacy.hpp>
+#elif (VISP_HAVE_OPENCV_VERSION >= 0x020101) // Require opencv >= 2.1.1
 #  include <opencv2/imgproc/imgproc.hpp>
 #  include <opencv2/features2d/features2d.hpp>
 #  include <opencv2/calib3d/calib3d.hpp>
